### ISSUE TITLE
Привязка адресов к провайдерам.

### ISSUE THREAD

**Issue by gruy** [gruy/crm#12]
Сделать связку адресов и провайдеров, чтобы адреса, где услуги провайдеров не осуществляются, не показывать в списке доступных поверхностей.

Что необходимо реализовать:

- [x] 1. Создать связь "Клиент-Адрес"
- [x] 2. Сделать страницу настройки связей "Клиент-Адрес"
- [x] 3. В списке поверхностей добавить фильтр "Зона покрытия"
- [x] 4. При создании "Заказа" выводить только те адреса, к которым привязан "Клиент".
- [x] 5. Колонка с провайдерами в таблице списка поверхностей.

**Comment by igorancho** [comment]
так, а где эта связь клиент и адрес?не нашёл как и где заполнить?

**Comment by igorancho** [comment]
можно ли и как реализовать возможность сортировки адресов с использованием нескольких зон покрытия
т.е. вывести итоговую таблицу адресов где будут отмечены например два или три провайдера, если они например присутствуют оба в одном и том же доме

**Comment by gruy** [comment]
Вариант: в таблице добавляется колонка для списка провайдеров и в ней показываем всех провайдеров на этой поверхности.

**Comment by igorancho** [comment]
да , такая колонка не помешает, не все районы ещё заполнили по провайдеру одному чтобы проверить поиск.
волжский заполнили, завтра попробую по нему поиски/выгрузки адресов разные поделать

**Comment by gruy** [comment]
Добавил колонку.

**Comment by igorancho** [comment]
http://nadomofone.ru/client/837/surfaces/
здесь в зоне покрытия все подъезды по нулям

**Comment by igorancho** [comment]
хорошо ещё чтобы можно было выбирать колличество строк показываемых на одной странице, как в поверхностях

**Comment by gruy** [comment]
Сделал.

**Comment by igorancho** [comment]
он показывает сколько домов, а нам нужно сколько поверхностей(подъездов) в зоне

**Comment by gruy** [comment]
Добавил вывод количества подъездов в зоне покрытия клиента

**Comment by igorancho** [comment]
http://nadomofone.ru/client/858/surfaces/
по волжскому почему то стало 0 подъездов, хотя забивали на этоу неделе, и там около 200 подъездов было

**Comment by igorancho** [comment]
тут такой же алгортитм как при добавлении в заказ? там у нас две кнопки, сперва добавить адреса, а потом ещё нужно сохранить.
но если бы мы не сохранили,тогда они бы сразу исчезли, а пару раз на неделе я проверял и по влж он выдавал зону присутствия

**Comment by gruy** [comment]
Они сохраняются при нажатии кнопки "Добавить адреса".

**Comment by igorancho** [comment]
адреса именно по волжскому были сохранены через кнопку добавить адреса.

**Comment by igorancho** [comment]
в колонке провайдеры сейчас указаны: Билайн влж, Домру влж
хотя использовать зону покрытия включена только для билайн влж, у домру влж она не включена и адреса даже не заполнены
http://nadomofone.ru/surface/?city=3&area=&street=&management=&has_stand=0&free=0&date_start=20.04.2022&date_end=20.04.2022&client=&client_surface=0

**Comment by gruy** [comment]
Исправлено.

У Домру влж указаны адреса в зоне покрытия, но отключена сама опция зоны покрытия.

**Comment by igorancho** [comment]
необходимо ещё добавить сортировку НЕБилайн, НЕДомру, НЕ Зона покрытия типа
например: я хочу выбрать адреса, которые свободные и которые не принадлежат к какойто зоне покрытия, сейчас я таку сортировку не могу сделать

**Comment by gruy** [comment]
Добавил исключение по выбранной зоне покрытия.
![image](https://user-images.githubusercontent.com/2075629/168065705-53418b50-59fe-481f-9cb9-143295b83a87.png)

**Comment by igorancho** [comment]
![image](https://user-images.githubusercontent.com/54850688/170697100-82abbf59-da29-4c8a-ab39-2cda2cbcf67f.png)
сделал выборку свободные и исключить зону покрытия билайн, всё равно попали адреса которые принадлежат билайн, справа это зона покрытия билайн, слева это то что выгрузилось в эксель,(свободно, исключить зону покрытия билайн)
http://nadomofone.ru/surface/?city=1&area=0&street=&management=0&has_stand=0&free=1&date_start=01.06.2022&date_end=04.06.2022&client=0&client_surface=837&client_surface_exclude=2

**Comment by gruy** [comment]
Добавил исключение по зоне покрытия в экспорт Excel
